airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/BufferManager.java

```diff
@@ -5,12 +5,17 @@
 package io.airbyte.integrations.destination_async;
 
 import io.airbyte.integrations.destination.buffered_stream_consumer.RecordSizeEstimator;
+import io.airbyte.integrations.destination_async.MemoryBoundedLinkedBlockingQueue.MemoryItem;
 import io.airbyte.protocol.models.v0.AirbyteMessage;
 import io.airbyte.protocol.models.v0.StreamDescriptor;
 import java.time.Instant;
 import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Stream;
 import lombok.extern.slf4j.Slf4j;
 
+// todo (cgardens) - why is buffer manager autocloseable?
```

Comment:
I added this autocloseable to close the buffers. was trying to debug the zombie thread issues.

Reply:
I don't think it helped.